### DISCUSSION TITLE
Optimize for target CAGR

### DISCUSSION THREAD

**Discussion by aliencaocao** [dcajasn/Riskfolio-Lib#114] [Q&A] [answered]
Hi, i want to optimize for minrisk but i need to meet (or exceed) a CAGR of 8.45%. I am using SLPM as risk measure. I have a few questions:
1. Is it correct for me to set port.lowerret = 0.0845/252, assuming my returns are daily?
2. If so, why is it that the optimized portfolio exceed the target CAGR by a large margin? Shouldnt it try to stick to the CAGR for lower risk?
3. What is the role of risk free rate here? Can I set it to a non 0 value and still expect the lowerret requirement to work as intended?
4. What is the 'minium accepted return' displayed in jupyter report? For this particular portfolio, it shows 0.0159% with footnote saying it is already annualized by multiplying 252. Shouldnt this be 8.45%?

Thank you

**Comment by dcajasn** [comment]
Hi @aliencaocao,

You can't set an objective for CAGR because optimization models use arithmetic returns. You can approximate CAGR using annualized arithmetic return. The constraint you are using is $\mu w \geq 0.0845/252$. For other questions, I recommend you to read the papers related to lower partial moments and Sharpe ratio optimization.

Best,
DANY

**Comment by aliencaocao** [comment]
May I know the difference between the Mean return and CAGR in the portfolio report?

And for minium accepted return, it seems to be the risk free rate, but am I not supposed to pass in the risk free rate normalized to daily? Because right now I assume the annual risk free rate to be 4%, and I do rf=0.04/252, but in the report, it is still reported as 0.04/252 but with a footnote saying it is already multiplied by 252 to annualize. This is very confusing to me. I expected it to be 4% after the annualized.

Latestly, if I want my portfolio to grow by 8.5% per year (with respective to the previous year), how should I set the arguments? I do not know the difference between the 'arithmetic returns' and CAGR, and I cannot find any info on converting one to the other from google.

**Comment by dcajasn** [comment]
Mean return is arithmetic mean return, CAGR is geometric return annualized this is a well known concept. In the case of partial moments riskfolio-lib takes $r_f$ as Minimum accepted return. I've just found the error in reports and I will fix it in next version.

On the other hand, you can't set a CAGR objective, you only can set an expected return objective based on the sample.

**Comment by aliencaocao** [comment]
So should I use rf=0.04/252 or just 0.04 if my returns are daily?
By "expected return objective based on the sample", do you mean the target return for the entire period of the returns df passed in?
E.g. I pass in last 5 years of data and I want to optimize for a 5 year total return of 50%. Do I set the lowerret to be 0.50/5/252?

Thank you for your patience

**Comment by dcajasn** [comment, marked as answer]
You must use $r_f=0.04/252$. On the other hand, I recommend you to study the different kind of return's concepts (arithmetic, geometric, logarithmic, CAGR, total return) because it seems that you don't know the difference among them.

**Comment by aliencaocao** [comment]
Alright, thank you for your help
